src/main/kotlin/racingcar/Application.kt

```diff
@@ -1,5 +1,125 @@
 package racingcar
 
+import camp.nextstep.edu.missionutils.Console
+import camp.nextstep.edu.missionutils.Randoms
+import kotlin.collections.all
+import kotlin.text.all
+
+var cars = mutableListOf<String>()
+var movingTime: Int = 0
+var carRunStatus = mutableMapOf<String, Int>()
+
 fun main() {
     // TODO: 프로그램 구현
+    game()
 }
+
+fun game() {
+    println("경주할 자동차 이름을 입력하세요.(이름은 쉼표(,) 기준으로 구분)")
+    carNameSetting()
+    println("시도할 횟수는 몇 회인가요?")
+    timeSetting()
+    println()
+    println("실행 결과")
+    stepControl()
+    winner()
+}
+
+fun input(): String {
+    return Console.readLine()
+}
+
+fun processCarNaming(): List<String> {
+    return input().split(",")
+}
+
+fun validateNamesLength(names: List<String>) {
+    if (!names.all { it.length <= 5 }) {
+        throw IllegalArgumentException("이름은 5자 이하여야 합니다.")
+    }
+    if (!names.all { it.isNotEmpty() }) {
+        throw IllegalArgumentException("이름을 비워둘 수 없습니다.")
+    }
+}
+
+fun validateNamesType(names: List<String>) {
+    for (name in names) {
+        validateNamesCharType(name)
+    }
+}
+
+private fun validateNamesCharType(name: String) {
+    if (!name.all { it.isLetter() }) {
+        throw IllegalArgumentException("이름은 알파벳만 입력 가능합니다.")
+    }
+}
+
+fun pickRandomNumber(): Int {
+    return Randoms.pickNumberInRange(0, 9)
+}
+
+fun timeSetting() {
+    val inputTime = input()
+    movingTime = inputTime.toIntOrNull()
+        ?: throw IllegalArgumentException("숫자만 가능합니다.")
+    if (movingTime <= 0) {
+        throw IllegalArgumentException("횟수는 1 이상이어야 합니다.")
+    }
+}
+
+fun stepControl() {
+    for (i in 1..movingTime) {
+        racingFourMoreRole()
+        currentResult()
+    }
+}
+
+fun racingFourMoreRole() {
+    for (carName in cars) {
+        judgeFourMoreRole(carName)
+    }
+}
+
+fun judgeFourMoreRole(car: String) {
+    val randomNumber = pickRandomNumber()
+    if (randomNumber >= 4) {
+        val currentStatus = carRunStatus.getOrDefault(car, 0)
+        carRunStatus[car] = currentStatus + 1
+    }
+}
+
+fun currentResult() {
+    for (carName in cars) {
+        print("$carName : ")
+        barCreator(carRunStatus[carName])
+    }
+    println()
+}
+
+fun barCreator(time: Int?) {
+    val bar = "-"
+    for (i in 1..time!!) {
+        print(bar)
+    }
+    println()
+}
```

Comment:
전진 횟수만큼 "-"를 출력할 때 `repeat()`을 활용하면 코드가 간결해질 것 같습니다!
```suggestion
fun currentResult() {
    for (carName in cars) {
        println("$carName : ${"-".repeat(carRunStatus[carName]!!)}")    
    }
}
```

Reply:
일단 리뷰 달아주셔서 정말 감사드린다는 말씀드리고 싶습니다!!
repeat 함수란 게 있었군요! 새롭게 배워가요!
그리고 class랑 파일로 정리해서 코드 작성하는 법에 대해 알아봐야겠어요!
함수명도 동사형으로 쓰는 게 정말 보기에 더 좋고 와닿는 거 같습니다.

정말 감사합니다!!